provider/datagen/src/transform/icuexport/uprops/enum_codepointtrie.rs

```diff
@@ -52,20 +53,74 @@ macro_rules! expand {
                     Ok(vec![Default::default()])
                 }
             }
+
+            impl DataProvider<$names_marker> for crate::DatagenProvider
+            {
+                fn load(&self, _: DataRequest) -> Result<DataResponse<$names_marker>, DataError> {
+                    let data = get_enumerated_prop(&self.source, $prop_name)
```

Comment:
Nit: Too much logic inside the macro. Please move this to a non-macro helper function and call the helper function.